### ISSUE TITLE
Error when running `./pipeline_t5.sh pretraining_experiments/pretraining_t5_ourtasks_nonsense`. Using `Python 3.6.9` and ran `pip install -r requirements.txt`

### ISSUE THREAD

**Issue by felixzli** [acmi-lab/pretraining-with-nonsense#1]
Hello,

I am trying to pretrain on the ourtasks-nonsense dataset with the configs `pretraining_experiments/pretraining_t5_ourtasks_nonsense.jsonnet`

I am using Python 3.6.9 because the README says that the code was tested using Python 3.6

**Attempt 1**

When I ran `pip install -r requirements.txt`
I got the below error
<img width="1347" alt="image" src="https://user-images.githubusercontent.com/31943554/147336909-1dd2e1c2-316a-4685-ade9-9a67045eba04.png">
So in requirements.txt I changed `torch==1.5.1` to just `torch` and then ran `pip install -r requirements.txt`

Then I ran: 
`cd pretraining_datasetgen`
`python ourtasks.py`
`allennlp train pretraining_experiments/pretraining_t5_ourtasks_nonsense.jsonnet --include-package t5 -s  pretraining_experiments/pretraining_t5_ourtasks_nonsense`

From the `allennlp` command, I get the error: 
<img width="1180" alt="image" src="https://user-images.githubusercontent.com/31943554/147340299-7d36dc06-8af0-483b-8518-81edf031c1ca.png">

**Attempt 2**

In requirements.txt I removed `torch==1.5.1` before running `pip install -r requirements.txt`. After this, I installed torch 1.5.1 by itself with `pip install torch==1.5.1`.

Then I ran: 
`cd pretraining_datasetgen`
`python ourtasks.py`
`allennlp train pretraining_experiments/pretraining_t5_ourtasks_nonsense.jsonnet --include-package t5 -s  pretraining_experiments/pretraining_t5_ourtasks_nonsense`

From the `allennlp` command, I get the error: 
<img width="1166" alt="image" src="https://user-images.githubusercontent.com/31943554/147338540-91f5548a-06b1-4a09-a530-6ac1b5ea48a5.png">

Thanks for any help

**Comment by kukrishna** [comment]
Hi,

Yes, pytorch complains when using .view instead of .reshape 
We simply followed the suggestion that the error provides.
Can you change the line 1154 in modeling_t5.py to 
`loss = loss_fct(lm_logits.reshape(-1, lm_logits.size(-1)), labels.reshape(-1))`

That fixed it for us.

Regarding the pytorch version, our codebase uses version 1.5.1, so we put it in the requirements.txt, but I wont be surprised if higher versions work okay too.

**Comment by felixzli** [comment]
Thanks that works! My bad, I should have tried following the suggestion the error provides myself before posting this. I didn't know how straightforward it is to change code of an installed Python module.
